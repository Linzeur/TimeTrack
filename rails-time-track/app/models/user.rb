class User < ApplicationRecord
  has_secure_password
  has_secure_token

  has_many :project_members
  has_many :projects, through: :project_members
  has_many :daily_logs, through: :project_members

  def self.valid_login?(email, password)
    user = find_by(email: email)
    user if user&.authenticate(password)
  end
end
