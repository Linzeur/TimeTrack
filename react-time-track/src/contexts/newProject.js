import React from "react";

const NewProjectContext = React.createContext();

function NewProjectProvider({ infoProject, children }) {
  return (
    <NewProjectContext.Provider value={infoProject}>
      {children}
    </NewProjectContext.Provider>
  );
}

function useNewProjectConsumer() {
  return React.useContext(NewProjectContext);
}

export { NewProjectProvider, useNewProjectConsumer };
