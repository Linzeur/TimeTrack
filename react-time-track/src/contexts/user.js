import React from "react";

const UserContext = React.createContext();

function UserProvider({ children }) {
  const [loggedIn, setLoggedIn] = React.useState(
    JSON.parse(localStorage.getItem("user")) || null
  );

  React.useEffect(() => {
    if (loggedIn) localStorage.setItem("user", JSON.stringify(loggedIn));
    else localStorage.removeItem("user");
  }, [loggedIn]);

  const onLogin = userData => setLoggedIn(userData);
  const onLogout = () => setLoggedIn(null);

  const value = {
    data: loggedIn,
    onLogin: onLogin,
    onLogout: onLogout
  };

  return <UserContext.Provider value={value}>{children}</UserContext.Provider>;
}

export { UserProvider, UserContext };
