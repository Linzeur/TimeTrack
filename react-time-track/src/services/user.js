import { apiUrl } from "../util";
// "http://localhost:3000/api/"

const API_USERS = `${apiUrl}users`;

async function userList() {
  const userList = await fetch(API_USERS, {
    method: "GET",
    credentials: "include",
    headers: {
      "Content-Type": "application/json"
    }
  });

  if (!userList.ok) {
    const { errors } = await userList.json();
    console.log(errors);
    throw new Error(errors);
  }

  return userList.json();
}

async function createUser(userData) {
  console.log(userData);
  const response = await fetch(API_USERS, {
    method: "POST",
    credentials: "include",
    headers: {
      "Content-Type": "application/json"
    },
    body: JSON.stringify(userData)
  });

  if (!response.ok) {
    const { errors } = await response.json();
    throw new Error(errors.message);
  }

  return response.json();
} 

async function editUser(userId) {
  console.log(userId);
  const response = await fetch(`${API_USERS}/${userId}`, {
    method: "PUT",
    credentials: "include",
    headers: {
      "Content-type": "application/json"
    }
  });

  if (!response.ok) {
    const {errors} = await response.json();
    throw new Error(errors.message);
  }

  return response.json();
}

async function getUser(userId) {
  console.log(userId);
  const response = await fetch(`${API_USERS}/${userId}`, {
    method: "GET",
    credentials: "include",
    headers: {
      "Content-Type": "application/json"
    }
  });

  if (!response.ok) {
    const { errors } = await response.json();
    throw new Error(errors.message);
  }

  return response.json();
} 

export { userList, createUser, getUser, editUser };
