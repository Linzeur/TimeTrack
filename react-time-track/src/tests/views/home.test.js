import React from "react";
import { render } from "@testing-library/react";

import { UserProvider, UserContext } from "../../contexts/user";
import Home from "../../views/home";
import AllProjects from "../../views/all-projects";

test("Home component", () => {
  const user = {
    id: 2,
    name: "Brayan Manager",
    email: "linzeur@hotmail.com",
    role: "Manager",
    rate: 4300
  };
  localStorage.setItem("user", JSON.stringify(user));
  const { asFragment } = render(
    <UserProvider>
      <UserContext.Consumer>
        {logged => (
          <Home>
            <AllProjects />
          </Home>
        )}
      </UserContext.Consumer>
    </UserProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
