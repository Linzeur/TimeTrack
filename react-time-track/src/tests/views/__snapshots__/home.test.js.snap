// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component 1`] = `
<DocumentFragment>
  <div
    class="css-14ytakr-Home"
  >
    <div
      class="css-1cnrnmv-Home"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
        />
      </svg>
    </div>
    <nav
      class="css-1jq6sci-NavBar"
    >
      <div
        class="css-kgnwp5-NavBar"
      >
        <span>
          Time Tracker
        </span>
        <span
          class="css-m0dyg1-NavBar"
          role="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 352 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
            />
          </svg>
        </span>
      </div>
      <div
        class="css-194f13j-NavBar"
      >
        <div
          class="css-s7vyd6-NavBar"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
            />
          </svg>
        </div>
        <div
          class="css-163lrdk-NavBar"
        >
          <span>
            Brayan Manager
          </span>
          <span>
            Manager
          </span>
        </div>
      </div>
      <div
        class="css-1wl6jgt-NavBar"
      >
        <a
          href="/mystatus"
        >
          <div
            class="css-1hpvyxb-NavBarItem"
            link="/mystatus"
          >
            My Status
          </div>
        </a>
        <a
          aria-current="page"
          href="/"
        >
          <div
            class="css-1hpvyxb-NavBarItem"
            link="/"
          >
            Projects
          </div>
        </a>
        <a
          href="/members"
        >
          <div
            class="css-1hpvyxb-NavBarItem"
            link="/members"
          >
            Members
          </div>
        </a>
        <a
          href="/history"
        >
          <div
            class="css-1hpvyxb-NavBarItem"
            link="/history"
          >
            History
          </div>
        </a>
      </div>
    </nav>
    <div
      class="css-71a3tm-Home"
      role="group"
      style="outline: none;"
      tabindex="-1"
    >
      <div
        class="css-nwfifz-Header"
      >
        <h2
          class="css-jud8en-Header"
        >
          Projects
        </h2>
        <div
          class="css-1jfpn6l-Header"
        >
          <a
            href="/daily-log"
          >
            <div
              class="css-1eavuz3-NavBarItem"
              link="/daily-log"
            >
              <div
                class="css-1hkss1w-Header"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z"
                  />
                </svg>
              </div>
              <span
                class="css-7dh1m0-Header"
              >
                Log
              </span>
            </div>
          </a>
          <div
            class="css-1jw2xof-Header"
          >
            <div
              class="css-9shpz4-Header"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                />
              </svg>
            </div>
            <div
              class="css-1p5qcu1-Header"
            >
              <span>
                Brayan Manager
              </span>
              <span>
                Manager
              </span>
            </div>
            <div
              class="css-1op67rn-Header"
            >
              <div
                aria-label="log out session"
                class="css-1c4fmxo-LogOut"
                role="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-1ugqmjg-Home"
    >
      <div>
        <h1
          class="css-a6rct0-Subtitle"
        >
          All Projects:
        </h1>
        <main>
          <section
            class="css-10oyw64-Section"
            role="list"
          />
        </main>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
